### PR TITLE
Add category model

### PR DESCRIPTION
**Pull request by denysokhrymovych**

### CODE REVIEW

**Review by DPotapov93 — APPROVED** [review]
Good job!

File: src/main/java/com/example/bookshop/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @GetMapping("/{id}")
+    @Operation(
+            summary = "Get a specific category by its id",
+            description = "Get a specific category by its id")
+    public CategoryResponseDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/CategoryController.java:63]:
```suggestion
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED) 
```

File: src/main/java/com/example/bookshop/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
+    @GetMapping("/{id}")
+    @Operation(
+            summary = "Get a specific category by its id",
+            description = "Get a specific category by its id")
+    public CategoryResponseDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/CategoryController.java]:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```

File: src/main/java/com/example/bookshop/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    @PostMapping
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    public CategoryResponseDto createCategory(@RequestBody @Valid CategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/CategoryController.java]:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```

File: src/main/java/com/example/bookshop/controller/CategoryController.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    @PutMapping("/{id}")
+    @Operation(summary = "Update a specific category", description = "Update a specific category")
+    public CategoryResponseDto updateCategory(
+            @PathVariable Long id,
+            @RequestBody @Valid CategoryRequestDto requestDto) {
+        return categoryService.update(id, requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/CategoryController.java]:
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```

File: src/main/java/com/example/bookshop/model/Category.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/model/Category.java:27]:
```suggestion
    @Column
    private String description;
```

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but there are some minor changes should be done!

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**